src/gpgservice.py

```diff
@@ -0,0 +1,69 @@
+import gnupg
+import json
+import pprint
+import os
+
+class GPGService:
+
+	def __init__(self):
+		super(GPGService, self).__init__()
+		self.gpg = gnupg.GPG(gnupghome ='homegpg')
+		with open('keys.json','a+') as dataFile:
+			if os.stat('keys.json').st_size is not 0:
+				self.keysFp = json.load(dataFile)
+			else:
+				self.keysFp = {}
+
+	def createKeys(self, domain):
+		self.inputData = self.gpg.gen_key_input(
+				name_email = domain,
+				passphrase = 'password')
+
+		key = self.gpg.gen_key(self.inputData)
+
+		self.keysFp[domain] = key.fingerprint
+		self.updateKeysFpFile()
+
+	def encodeMessage(self, message, senderDomain):
+		return str(self.gpg.encrypt(message, senderDomain))
+
+	def decodeMessage(self, message):
+		decryptedData = self.gpg.decrypt(message, passphrase ='password')
+		return decryptedData.ok
+
+	def printAllKeys(self):
```

Comment:
We also need a method to programatically retrieve the keys (only public ones though), not just print them.